formats/avro/src/main/java/io/cloudevents/avro/AvroCloudEventDataWrapper.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2018-Present The CloudEvents Authors
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package io.cloudevents.avro;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+import io.cloudevents.AvroCloudEventData;
+import io.cloudevents.CloudEventData;
+
+/**
+ * Encode JSON style cloudevent data into Avro format.
+ *
+ */
+public class AvroCloudEventDataWrapper implements CloudEventData {
+
+    private AvroCloudEventData avroCloudEventData;
+
+    /**
+     * Wraps a JSON object-like data structure.
+     */
+    public AvroCloudEventDataWrapper(Map<String, Object> data) {
+        avroCloudEventData = new AvroCloudEventData();
+        avroCloudEventData.setValue(data);
+    }
```

Comment:
The reason I'm using a map here is I tried to avoid exposing generated data type in a public api. The `AvroCloudEventData` contains some Avro serialization logic.

Reply:
OK, sounds good!

Can we add only the `Objects.requireNonNull` check?

```suggestion
    public AvroCloudEventDataWrapper(Map<String, Object> data) {
        avroCloudEventData = new AvroCloudEventData();
        avroCloudEventData.setValue(Objects.requireNonNull(data));
    }
```